cmd/logmonitor/db.go

```diff
@@ -0,0 +1,118 @@
+// Copyright 2018 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/golang/glog"
+	"github.com/google/trillian/types"
+	"github.com/jinzhu/gorm"
+
+	// Include all currently supported gorm dialects
+	_ "github.com/jinzhu/gorm/dialects/mssql"
+	_ "github.com/jinzhu/gorm/dialects/mysql"
+	_ "github.com/jinzhu/gorm/dialects/postgres"
+	_ "github.com/jinzhu/gorm/dialects/sqlite"
+)
+
+// initialEmptyRoot represents an empty trusted root and is used when no
+// information is known about the log being monitored.
+var initialEmptyRoot TrustedLogRoot
+
+func init() {
+	// Set up initialEmptyRoot
```

Comment:
Please add a TODO to save and restore the last-verified root.

Reply:
This is already the case.

Reply:
I'll update the comment to clarify this point.